### PR TITLE
Add pre-commit hook & fix shell script lint errors

### PR DESCRIPTION
**Pull request by letmaik**
This PR is copying some of the pre-commit scripts from OE. Currently I only enabled linting of shell scripts using `shellcheck` and fixed corresponding issues. There's also code formatting using `clang-format` but since this is much more disruptive I will enable that (and re-format code) once we agreed on the right strategic time for it.

cc @wintersteiger

### CODE REVIEW

File: src/sched/lthread.c [hunk 1]
```diff
@@ -27,6 +27,8 @@
  *
  */
 
+// clang-format off
```
Comment (prp) [inline src/sched/lthread.c]: I don’t see the point of keeping different formatting here, as we are not tracking any upstream.

**Review by prp — APPROVED** [review]
Thanks, LGTM. I left one comment.

**Comment by prp** [comment]
Oops, sgx-lkl-disk needs fixing.

**Comment by letmaik** [comment]
> 
> 
> Oops, sgx-lkl-disk needs fixing.

Yeah, I'll mark this as draft until CI is passing.

File: src/sched/lthread.c (hunk 1 above)
Comment (letmaik) [reply src/sched/lthread.c]: OK makes sense, the license header mislead me.

**Comment by wintersteiger** [comment]
This is great, many thanks @letmaik!

**Pull request merged**